### DISCUSSION TITLE
PESTPP-iES and PWTADJ1?

### DISCUSSION THREAD

**Discussion by jrwbell** [usgs/pestpp#216] [Q&A] [answered]
Hi.  I'm a new user of PESTPP-iES, but have been a 'vanilla' PEST user for a long time now.  I am used to using PWTADJ1 to 'balance' the objective function.  My approach with PESTPP-iES, to allow me to 'balance' my objective function, was to generate my own observation ensemble via PEST's RANDOBS.  I then fed this into PESTPP-iES via "++ ies_observation_ensemble(calib1_init_obs_ensemble_QSS.jcb)".

When looking at how this model is progressing though, there are "calib2b.obs+noise.jcb", which was written at the start of the run, then "calib2b.0.obs.jcb" and then "calib2b.1.obs.jcb".  Doing a binary compare, they are all different.  My question is does this matter or not?  At the end of the day, it is fine, as far as I am concerned, if the 'noise' is changing, but, if PESTPP-iES is using the weights in the .PST to inform the choice of that noise, then they are the wrong weights, since they've been adjusted via PWTADJ1.

It may be that I can't use PWTADJ1 because PESTPP-iES needs those weights at each iteration.  If that is the case, what does a PESTPP-iES user do about the fact that an objective function is often not 'balanced', due to different units, number of observations etc?

**Comment by cnicol-gwlogic** [comment, marked as answer]
Hi @jrwbell ,

Jeremy will be better placed to answer this than I am, but: 
* I believe that case.obs+noise.jcb is your observation ensemble, with noise added. I.e. straight from your provided randobs ensemble.
* Obs.*.jcb/.csv naming convention actually refers to the model output equivalents of your observations. The obs.0.jcb and obs.1.jcb etc are those for each iteration, with .0.jcb being your prior, and .1.jcb being the outcome of the first parameter upgrade iteration. 

So the ptwadj1 issue you mention is not an issue, your setup should be working as you intended.

See [https://github.com/usgs/pestpp/blob/master/documentation/pestpp_users_manual.md#931-csv-output-files]()  when in doubt. Hope this helps.

**Reply by wkitlasten** [reply]
You can generate an obs+noise ensemble from you original weights to pass to pestpp. Then adjust your weights before the run.

There are utilities in pyemu to do all that. I guess it all boils down to how the problem is behaving and how it impacts the predictions.

**Comment by rhugman** [comment]
Previous replies have covered the question, but if you’d like a tutorial on using IES and PWADJ utilities see the the GMDSI tutorials here: https://gmdsi.org/education/tutorials/

They cover this issue of obs noise vs obs weighting schemes.

**Comment by jrwbell** [comment]
Thanks all.  A brave new world.
